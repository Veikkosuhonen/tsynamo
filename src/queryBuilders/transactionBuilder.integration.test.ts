import { DynamoDBDocumentClient } from "@aws-sdk/lib-dynamodb";
import { DDB } from "../../test/testFixture";
import { getDDBClientFor, startDDBTestContainer } from "../../test/testUtil";
import { Tsynamo } from "../index";

describe("TransactionBuilder", () => {
  let tsynamoClient: Tsynamo<DDB>;
  let ddbClient: DynamoDBDocumentClient;

  beforeAll(async () => {
    const testContainer = await startDDBTestContainer();
    ddbClient = await getDDBClientFor(testContainer);

    tsynamoClient = new Tsynamo<DDB>({
      ddbClient,
    });
  });

  it("handles puts", async () => {
    const trx = tsynamoClient.createTransaction();

    trx.addItem({
      Put: tsynamoClient
        .putItem("myTable")
        .item({ userId: "9999", dataTimestamp: 1 }),
    });

    trx.addItem({
      Put: tsynamoClient
        .putItem("myTable")
        .item({ userId: "9999", dataTimestamp: 2 }),
    });

    await trx.execute();

    const result = await tsynamoClient
      .query("myTable")
      .keyCondition("userId", "=", "9999")
      .execute();

    expect(result).toMatchSnapshot();
  });
});
