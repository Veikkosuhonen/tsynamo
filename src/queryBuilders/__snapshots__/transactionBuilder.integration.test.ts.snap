// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TransactionBuilder > handles puts 1`] = `
[
  {
    "dataTimestamp": 1,
    "userId": "9999",
  },
  {
    "dataTimestamp": 2,
    "userId": "9999",
  },
]
`;
